src/core/Global.ttslua

```diff
@@ -619,176 +638,285 @@ end
 -- Content Importing and XML functions
 ---------------------------------------------------------
 
-function onClick_refreshList()
-  local request = WebRequest.get(SOURCE_REPO .. '/library.json', completed_list_update)
-  requestObj = request
-  startLuaCoroutine(Global, 'downloadCoroutine')
+-- forwards the requested content type to the update function and sets highlight to clicked tab
+---@param tabId String Id of the clicked tab
+function onClick_tab(_, _, tabId)
+  for listId, listContent in pairs(tabIdTable) do
+    if listId == tabId then
+      UI.setClass(listId, 'downloadTab activeTab')
+      contentToShow = listContent
+    else
+      UI.setClass(listId, 'downloadTab')
+    end
+  end
+  currentListItem = 1
+  updateDownloadItemList()
 end
 
-function onClick_select(player, params)
-  params = JSON.decode(urldecode(params))
-  local url = SOURCE_REPO .. '/' .. params.url
-  local request = WebRequest.get(url, function (request) complete_obj_download(request, params) end )
-  requestObj = request
+-- click function for the items in the download window
+-- updates backgroundcolor for row panel and fontcolor for list item
+function onClick_select(_, _, identificationKey)
+  UI.setAttribute("panel" .. currentListItem, "color", "clear")
+  UI.setAttribute(contentToShow .. "_" .. currentListItem, "color", "white")
+  
+  -- parses the identification key (contentToShow_currentListItem)
+  if identificationKey then
+    contentToShow = nil
+    currentListItem = nil
+    for str in string.gmatch(identificationKey, "([^_]+)") do
+      if not contentToShow then
+        -- grab the first part to know the content type
+        contentToShow = str
+      else
+        -- get the index
+        currentListItem = tonumber(str)
+        break
+      end
+    end
+  end
+
+  UI.setAttribute("panel" .. currentListItem, "color", "grey")
+  UI.setAttribute(contentToShow .. "_" .. currentListItem, "color", "black")
+  updatePreviewWindow()
+end
+
+-- click function for the download button in the preview window
+function onClick_download()
```

Comment:
Should definitely add the click sound effect to this one

Reply:
Done!